src/idpi/product.py

```diff
@@ -0,0 +1,40 @@
+"""Product base classes."""
+
+# Standard library
+import dataclasses as dc
+from abc import ABCMeta
+from abc import abstractmethod
+from functools import partial
+
+# Third-party
+import dask
+
+
+@dc.dataclass
+class ProductDescriptor:
+    input_fields: list[str]
```

Comment:
I was thinking about a different design but I can try to make a PR about it. This is fine as it is now